In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from tqdm import tqdm
import missingno as msno
import seaborn as sns
import scipy.stats as st
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
import xgboost as xg
from lightgbm import LGBMClassifier
from catboost import CatBoostClassifier
from sklearn.metrics import accuracy_score, f1_score
from eli5.sklearn import PermutationImportance
from eli5 import show_weights
from sklearn.tree import DecisionTreeClassifier
from bayes_opt import BayesianOptimization

In [2]:
#Загрузим данные из файла.
df = pd.read_excel('/home/rik/Рабочий стол/МИФИ/учеба/мо/curse/data.xlsx', index_col=0)
df.head()

IC50, mM    CC50, mM         SI  MaxAbsEStateIndex  MaxEStateIndex  \
0    6.239374  175.482382  28.125000           5.094096        5.094096   
1    0.771831    5.402819   7.000000           3.961417        3.961417   
2  223.808778  161.142320   0.720000           2.627117        2.627117   
3    1.705624  107.855654  63.235294           5.097360        5.097360   
4  107.131532  139.270991   1.300000           5.150510        5.150510   

   MinAbsEStateIndex  MinEStateIndex       qed        SPS    MolWt  ...  \
0           0.387225        0.387225  0.417362  42.928571  384.652  ...   
1           0.533868        0.533868  0.462473  45.214286  388.684  ...   
2           0.543231        0.543231  0.260923  42.187500  446.808  ...   
3           0.390603        0.390603  0.377846  41.862069  398.679  ...   
4           0.270476        0.270476  0.429038  36.514286  466.713  ...   

   fr_sulfide  fr_sulfonamd  fr_sulfone  fr_term_acetylene  fr_tetrazole  \
0           0             0           0                  0             0   
1           0             0           0                  0             0   
2           0             0           0                  0             0   
3           0             0           0                  0             0   
4           0             0           0                  0             0   

   fr_thiazole  fr_thiocyan  fr_thiophene  fr_unbrch_alkane  fr_urea  
0            0            0             0                 3        0  
1            0            0             0                 3        0  
2            0            0             0                 3        0  
3            0            0             0                 4        0  
4            0            0             0                 0        0  

[5 rows x 213 columns]

In [3]:
print("train.shape = {} rows, {} cols".format(*df.shape))

train.shape = 1001 rows, 213 cols


# Предобработка данных.

In [4]:
def transform(df): 
    discrete_feature = [feature for feature in df.columns if len(df[feature].unique())<25]
    count_discrete_feature = {}
    for i in tqdm(discrete_feature):
        cnt = 0
        for j in df[i]:
            if j != 0:
                cnt += 1
        count_discrete_feature[f'{i}'] = cnt
    count_discrete_feature = pd.Series(count_discrete_feature)
    df = df.drop(columns=count_discrete_feature[count_discrete_feature == 0].index.to_list())
    df = df.dropna()
    feature_del = ['BertzCT', 'Chi0', 'Chi0v', 'Chi1', 'Chi1n', 'Chi1v', 'Chi2v', 'Chi3n', 'Chi3v', 'Chi4n', 'Chi4v',
     'ExactMolWt', 'FpDensityMorgan1', 'FpDensityMorgan3', 'HeavyAtomCount', 'HeavyAtomMolWt', 'Kappa1', 'Kappa2',
     'Kappa3', 'LabuteASA', 'MaxAbsEStateIndex', 'MaxAbsPartialCharge', 'MolMR', 'MolWt', 'NHOHCount', 'NumAromaticCarbocycles',
     'NumHAcceptors', 'NumHeteroatoms', 'NumSaturatedCarbocycles', 'NumValenceElectrons', 'SMR_VSA4', 'SMR_VSA9',
     'SlogP_VSA6', 'TPSA', 'VSA_EState2', 'VSA_EState3', 'VSA_EState6', 'fr_Al_OH_noTert', 'fr_COO', 'fr_COO2',
     'fr_C_O_noCOO', 'fr_Nhpyrrole', 'fr_benzene', 'fr_nitro_arom_nonortho', 'fr_phenol', 'fr_phenol_noOrthoHbond',
    'MinAbsPartialCharge', 'Chi2n']
    df = df.drop(columns=feature_del)
    df =  df[df['SI'] < 250]
    
    return df

In [5]:
df = transform(df)

100%|███████████████████████████████████████| 110/110 [00:00<00:00, 4323.21it/s]


In [6]:
val_median = df['SI'].median()

In [7]:
df['SI'] = df['SI'].apply(lambda x: 0 if x < val_median else 1)
df['SI'].head()

0    1
1    1
2    0
3    1
4    0
Name: SI, dtype: int64

## Обучение нескольких моделей, их сравнение. 

In [8]:
target = df['SI']
data = df.drop(['CC50, mM','IC50, mM','SI'], axis=1)

In [9]:
x_train, x_valid = train_test_split(
    data, train_size=0.8, random_state=1
)
y_train, y_valid = train_test_split(
    target, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid.shape))

x_train.shape = 776 rows, 144 cols
x_valid.shape = 195 rows, 144 cols


In [10]:
# Создадим модель дерева решений
tree = DecisionTreeClassifier(random_state=27)
# Создадим модель CatBoost
model = CatBoostClassifier(random_state=27)
# Создадим модель lightgbm
model_LGBM = LGBMClassifier(random_state=27)
# Создадим модель xgboost
xgb_r = xg.XGBClassifier() 
# обучение моделей
tree.fit(x_train, y_train)
model.fit(x_train, y_train, verbose=0)
model_LGBM.fit(x_train, y_train, verbose=0)
xgb_r.fit(x_train, y_train, verbose=0)
# предсказание ответов для тренеровочной выборки
y_pred_tree_train = tree.predict(x_train)
y_pred_сat_Boost_train = model.predict(x_train)
y_pred_LGBM_train = model_LGBM.predict(x_train)
y_pred_xgb_train = xgb_r.predict(x_train)
# предсказание ответов для тестовой выборки
y_pred_tree = tree.predict(x_valid)
y_pred_сat_Boost = model.predict(x_valid)
y_pred_LGBM = model_LGBM.predict(x_valid)
y_pred_xgb = xgb_r.predict(x_valid)

print("Train tree f1_score: ", f1_score(y_train, y_pred_tree_train))
print("Train tree accuracy_score: ", accuracy_score(y_train, y_pred_tree_train))
print("Validation tree f1_score: ", f1_score(y_valid, y_pred_tree))
print("Validation tree accuracy_score: ", accuracy_score(y_valid, y_pred_tree))
print('*' * 50)
print("Train сat_Boost f1_score: ", f1_score(y_train, y_pred_сat_Boost_train))
print("Train сat_Boost accuracy_score: ", accuracy_score(y_train, y_pred_сat_Boost_train))
print("Validation сat_Boost f1_score: ", f1_score(y_valid, y_pred_сat_Boost))
print("Validation сat_Boost accuracy_score: ", accuracy_score(y_valid, y_pred_сat_Boost))
print('*' * 50)
print("Train LGBM f1_score: ", f1_score(y_train, y_pred_LGBM_train))
print("Train LGBM accuracy_score: ", accuracy_score(y_train, y_pred_LGBM_train))
print("Validation LGBM f1_score: ", f1_score(y_valid, y_pred_LGBM))
print("Validation LGBM accuracy_score: ", accuracy_score(y_valid, y_pred_LGBM))
print('*' * 50)
print("Train xgb f1_score: ", f1_score(y_train, y_pred_xgb_train))
print("Train xgb accuracy_score: ", accuracy_score(y_train, y_pred_xgb_train))
print("Validation xgb f1_score: ", f1_score(y_valid, y_pred_xgb))
print("Validation xgb accuracy_score: ", accuracy_score(y_valid, y_pred_xgb))

/usr/local/lib/python3.8/dist-packages/lightgbm/sklearn.py:736: UserWarning: 'verbose' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
  _log_warning("'verbose' argument is deprecated and will be removed in a future release of LightGBM. "


Train tree f1_score:  0.9461235216819974
Train tree accuracy_score:  0.9471649484536082
Validation tree f1_score:  0.5894736842105263
Validation tree accuracy_score:  0.6
**************************************************
Train сat_Boost f1_score:  0.905852417302799
Train сat_Boost accuracy_score:  0.904639175257732
Validation сat_Boost f1_score:  0.6494845360824741
Validation сat_Boost accuracy_score:  0.6512820512820513
**************************************************
Train LGBM f1_score:  0.9472329472329473
Train LGBM accuracy_score:  0.9471649484536082
Validation LGBM f1_score:  0.6564102564102564
Validation LGBM accuracy_score:  0.6564102564102564
**************************************************
Train xgb f1_score:  0.9475032010243278
Train xgb accuracy_score:  0.9471649484536082
Validation xgb f1_score:  0.606060606060606
Validation xgb accuracy_score:  0.6


Из приведенных данных выше мы видим низкое и среднее качество моделей, а также переобучение. Наилучшие результаты показали модели сat_Boost и LGBM. С ними и продолжим работать.

# Подбор признаков, их анализ и оценка важности.

**CatBoost**

In [11]:
col = x_train.columns.tolist()
estimator_cat_boost = PermutationImportance(
    model, scoring= "f1", random_state=27
)
importance_cat_boost = estimator_cat_boost.fit(x_train, y_train)

In [12]:
show_weights(importance_cat_boost, feature_names=col, top=x_train.shape[1])

Оставим только те признаки которые которые улучшают работу модели. И посмотрим как изменится метрика

In [13]:
feature_cat_boost = ['qed','PEOE_VSA6','MinAbsEStateIndex','MinPartialCharge','PEOE_VSA9','EState_VSA4','PEOE_VSA7',
'VSA_EState1','FpDensityMorgan2','fr_aryl_methyl','Ipc','BalabanJ','AvgIpc','EState_VSA5','EState_VSA6','EState_VSA7',
'VSA_EState5','VSA_EState10','fr_para_hydroxylation','VSA_EState9','Chi0n','SlogP_VSA12','PEOE_VSA3','NumAliphaticHeterocycles',
'PEOE_VSA8','VSA_EState4','SlogP_VSA4','EState_VSA3','fr_term_acetylene','fr_Ar_N','NOCount','NumSaturatedRings',
'SlogP_VSA1','fr_NH1','NumHDonors','PEOE_VSA5','fr_piperzine','EState_VSA2','SMR_VSA6','EState_VSA8','MaxEStateIndex',
'BCUT2D_LOGPHI','fr_halogen','fr_Ndealkylation1','SMR_VSA3','fr_methoxy','NumRotatableBonds','PEOE_VSA14',
'SlogP_VSA7','fr_NH0','fr_quatN','fr_pyridine','fr_sulfide','FractionCSP3','HallKierAlpha','BCUT2D_MWHI','PEOE_VSA13',
'SMR_VSA5','fr_aniline','NumSaturatedHeterocycles']

In [14]:
target = df['SI']
data_cb = data[feature_cat_boost]

In [15]:
x_train_cb, x_valid_cb = train_test_split(
    data_cb, train_size=0.8, random_state=1
)
y_train_cb, y_valid_cb = train_test_split(
    target, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train_cb.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid_cb.shape))

x_train.shape = 776 rows, 60 cols
x_valid.shape = 195 rows, 60 cols


In [16]:
# Создадим модель CatBoost
model = CatBoostClassifier(random_state=27)
# обучение моделей
model.fit(x_train_cb, y_train_cb, verbose=0)
# предсказание ответов для тренеровочной выборки
y_pred_сat_Boost_train = model.predict(x_train_cb)
# предсказание ответов для тестовой выборки
y_pred_сat_Boost = model.predict(x_valid_cb)


print("Train сat_Boost f1_score: ", f1_score(y_train_cb, y_pred_сat_Boost_train))
print("Train сat_Boost accuracy_score: ", accuracy_score(y_train_cb, y_pred_сat_Boost_train))
print("Validation сat_Boost f1_score: ", f1_score(y_valid_cb, y_pred_сat_Boost))
print("Validation сat_Boost accuracy_score: ", accuracy_score(y_valid_cb, y_pred_сat_Boost))


Train сat_Boost f1_score:  0.9104859335038362
Train сat_Boost accuracy_score:  0.9097938144329897
Validation сat_Boost f1_score:  0.6632124352331605
Validation сat_Boost accuracy_score:  0.6666666666666666


Отобрав признаки мы улучшили метрики для модели CatBoost на вадидации f1_score: 0.66, accuracy_score: 0.67

**LGBM**

In [17]:
col = x_train.columns.tolist()
estimator_LGBM = PermutationImportance(
    model_LGBM, scoring= "f1", random_state=27
)
importance_LGBM = estimator_LGBM.fit(x_train, y_train)


In [18]:
show_weights(importance_LGBM, feature_names=col, top=x_train.shape[1])

Оставим только те признаки которые которые улучшают работу модели. И посмотрим как изменится метрика

In [19]:
feature_LGBM = ['HallKierAlpha','BCUT2D_MRLOW','EState_VSA4','BCUT2D_MRHI','MaxEStateIndex','MinPartialCharge',
'MinEStateIndex','NumRotatableBonds','PEOE_VSA6','PEOE_VSA7','PEOE_VSA9','BCUT2D_MWHI','PEOE_VSA8','VSA_EState4','SMR_VSA1',
'MaxPartialCharge','EState_VSA2','PEOE_VSA2','PEOE_VSA1','AvgIpc','EState_VSA9','Ipc','EState_VSA3','EState_VSA1',
'SlogP_VSA10','BCUT2D_LOGPHI','BalabanJ','fr_Ar_N','qed','fr_NH0','NumSaturatedRings','BCUT2D_LOGPLOW',
'NumAliphaticHeterocycles','VSA_EState5','NumHDonors','SMR_VSA6','MolLogP','NumSaturatedHeterocycles',
'fr_allylic_oxid','VSA_EState7','SlogP_VSA5','SMR_VSA10','PEOE_VSA10','PEOE_VSA4','Chi0n','VSA_EState9',
'BCUT2D_MWLOW','EState_VSA6','PEOE_VSA3','BCUT2D_CHGHI','NOCount','NumAliphaticRings','fr_C_S','EState_VSA7',
'PEOE_VSA13','VSA_EState10','fr_alkyl_carbamate','fr_ArN','fr_unbrch_alkane','fr_Ar_COO','fr_alkyl_halide','fr_aniline',
'fr_Al_COO','fr_bicyclic']

In [20]:
target_LGBM = df['SI']
data_LGBM = data[feature_LGBM]

In [21]:
x_train_LGBM, x_valid_LGBM = train_test_split(
    data_LGBM, train_size=0.8, random_state=1
)
y_train_LGBM, y_valid_LGBM = train_test_split(
    target_LGBM, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train_LGBM.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid_LGBM.shape))

x_train.shape = 776 rows, 64 cols
x_valid.shape = 195 rows, 64 cols


In [22]:
model_LGBM = LGBMClassifier(random_state=27)
model_LGBM.fit(x_train_LGBM, y_train_LGBM)
# предсказание ответов для тренеровочной выборки
y_pred_LGBM_train = model_LGBM.predict(x_train_LGBM)
# предсказание ответов для тестовой выборки
y_pred_LGBM = model_LGBM.predict(x_valid_LGBM)

print("Train LGBM f1_score: ", f1_score(y_train_LGBM, y_pred_LGBM_train))
print("Train LGBM accuracy_score: ", accuracy_score(y_train_LGBM, y_pred_LGBM_train))
print("Validation LGBM f1_score: ", f1_score(y_valid_LGBM, y_pred_LGBM))
print("Validation LGBM accuracy_score: ", accuracy_score(y_valid_LGBM, y_pred_LGBM))

Train LGBM f1_score:  0.9470967741935484
Train LGBM accuracy_score:  0.9471649484536082
Validation LGBM f1_score:  0.6699507389162561
Validation LGBM accuracy_score:  0.6564102564102564


Отобрав признаки мы улучшили метрики для модели LGBMClassifier на вадидации f1_score: 0.67, accuracy_score: 0.66.

# Подбор гиперпараметров

**CatBoost**

In [23]:
def catboost_cv(depth, learning_rate,n_estimators):
    depth = int(depth)
    n_estimators = int(n_estimators)
    model = CatBoostClassifier(
        depth=depth,
        learning_rate=learning_rate,
        n_estimators=n_estimators,
        eval_metric  = 'F1',
        random_seed=27
    )
    model.fit(x_train_cb, y_train_cb, verbose=100)

    return f1_score(y_valid_cb, model.predict(x_valid_cb))

In [24]:
param_space = {
    'depth': (2, 10),             
    'learning_rate': (0.001, 0.3),
    'n_estimators': (50, 1000)
}

bayesian_opt = BayesianOptimization(
    f=catboost_cv, pbounds=param_space, random_state=27)
bayesian_opt.maximize(init_points=5, n_iter=20)
results = pd.DataFrame(bayesian_opt.res)
results.sort_values(by='target', ascending=False, inplace=True)

|   iter    |  target   |   depth   | learni... | n_esti... |
-------------------------------------------------------------
0:	learn: 0.5890805	total: 3.35ms	remaining: 2.5s
100:	learn: 0.9420849	total: 305ms	remaining: 1.95s
200:	learn: 0.9476373	total: 609ms	remaining: 1.66s
300:	learn: 0.9476373	total: 922ms	remaining: 1.37s
400:	learn: 0.9472329	total: 1.23s	remaining: 1.06s
500:	learn: 0.9477707	total: 1.52s	remaining: 752ms
600:	learn: 0.9473684	total: 1.82s	remaining: 445ms
700:	learn: 0.9479034	total: 2.12s	remaining: 142ms
747:	learn: 0.9479034	total: 2.26s	remaining: 0us
| 1         | 0.6392    | 5.406     | 0.2446    | 748.6     |
0:	learn: 0.6739130	total: 68.6ms	remaining: 1m 7s
100:	learn: 0.9450830	total: 1.76s	remaining: 15.3s
200:	learn: 0.9476373	total: 3.45s	remaining: 13.4s
300:	learn: 0.9473684	total: 5.1s	remaining: 11.5s
400:	learn: 0.9476373	total: 6.8s	remaining: 9.82s
500:	learn: 0.9476373	total: 8.57s	remaining: 8.2s
600:	learn: 0.9476373	total: 10.2s	remaini

200:	learn: 0.9475032	total: 990ms	remaining: 1.6s
300:	learn: 0.9475032	total: 1.47s	remaining: 1.09s
400:	learn: 0.9475032	total: 1.96s	remaining: 608ms
500:	learn: 0.9473684	total: 2.44s	remaining: 117ms
524:	learn: 0.9476373	total: 2.56s	remaining: 0us
| 16        | 0.6633    | 6.347     | 0.244     | 525.8     |
0:	learn: 0.6630872	total: 9.11ms	remaining: 4.11s
100:	learn: 0.9475032	total: 1.02s	remaining: 3.54s
200:	learn: 0.9476373	total: 1.89s	remaining: 2.37s
300:	learn: 0.9477707	total: 2.74s	remaining: 1.38s
400:	learn: 0.9479034	total: 3.58s	remaining: 456ms
451:	learn: 0.9479034	total: 4.03s	remaining: 0us
| 17        | 0.6468    | 7.069     | 0.1594    | 452.0     |
0:	learn: 0.5890805	total: 5.49ms	remaining: 2.63s
100:	learn: 0.9462916	total: 336ms	remaining: 1.26s
200:	learn: 0.9476373	total: 671ms	remaining: 928ms
300:	learn: 0.9475032	total: 1.06s	remaining: 629ms
400:	learn: 0.9476373	total: 1.37s	remaining: 267ms
478:	learn: 0.9479034	total: 1.62s	remaining: 0us
|

In [25]:
best_hyperparameters = bayesian_opt.max
best_hyperparameters['params'] = {param: int(value) if param in [
    'depth', 'iterations', 'l2_leaf_reg'] else value for param, value in best_hyperparameters['params'].items()}
print("Best hyperparameters:", best_hyperparameters['params'])
print(f"Best R-squared Score: {best_hyperparameters['target']:.4f}")

Best hyperparameters: {'depth': 4, 'learning_rate': 0.1415804717945562, 'n_estimators': 775.6441966875759}
Best R-squared Score: 0.6736


In [26]:
params = {
    'depth': 4,             
    'learning_rate': 0.1415804717945562, 
    'n_estimators': 776
}
model = CatBoostClassifier(**params, random_state=27)
model.fit(x_train_cb, y_train_cb, verbose=100)
y_pred_cb = model.predict(x_train_cb)
y_val_cb = model.predict(x_valid_cb)
print("Train сat_Boost f1_score: ", f1_score(y_train_cb, y_pred_cb))
print("Train сat_Boost accuracy_score: ", accuracy_score(y_train_cb, y_pred_cb))
print("Validation сat_Boost f1_score: ", f1_score(y_valid_cb, y_val_cb))
print("Validation сat_Boost accuracy_score: ", accuracy_score(y_valid_cb, y_val_cb))

0:	learn: 0.6819893	total: 3.23ms	remaining: 2.5s
100:	learn: 0.3492358	total: 207ms	remaining: 1.38s
200:	learn: 0.2267826	total: 411ms	remaining: 1.17s
300:	learn: 0.1719675	total: 646ms	remaining: 1.02s
400:	learn: 0.1404115	total: 861ms	remaining: 805ms
500:	learn: 0.1243704	total: 1.08s	remaining: 592ms
600:	learn: 0.1153113	total: 1.32s	remaining: 384ms
700:	learn: 0.1087095	total: 1.56s	remaining: 167ms
775:	learn: 0.1059704	total: 1.73s	remaining: 0us
Train сat_Boost f1_score:  0.9472329472329473
Train сat_Boost accuracy_score:  0.9471649484536082
Validation сat_Boost f1_score:  0.6735751295336788
Validation сat_Boost accuracy_score:  0.676923076923077


После подбора гиперпараметров мы улучшили метрики модели CatBoost на вадидации f1_score: 0.67, accuracy_score: 0.68.

**LGBM**

In [27]:
def LGBM_cv(max_depth, learning_rate, n_estimators):
    max_depth = int(max_depth)
    n_estimators = int(n_estimators)
    model = LGBMClassifier(
        max_depth=max_depth,
        learning_rate=learning_rate,
        n_estimators=n_estimators,          
        eval_metric  = 'F1',
        random_seed=27
    )
    model.fit(x_train_LGBM, y_train_LGBM, eval_set = (x_valid_LGBM, y_valid_LGBM))
    return f1_score(y_valid_LGBM, model.predict(x_valid_LGBM))

In [28]:
param_space = {
    'max_depth': (2, 10),            
    'learning_rate': (0.001, 0.3),
    'n_estimators': (50, 1000)
}

bayesian_opt = BayesianOptimization(
    f=LGBM_cv, pbounds=param_space, random_state=27)
bayesian_opt.maximize(init_points=5, n_iter=25)
results = pd.DataFrame(bayesian_opt.res)
results.sort_values(by='target', ascending=False, inplace=True)

|   iter    |  target   | learni... | max_depth | n_esti... |
-------------------------------------------------------------
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.677618
[2]	valid_0's binary_logloss: 0.665184
[3]	valid_0's binary_logloss: 0.654325
[4]	valid_0's binary_logloss: 0.652278
[5]	valid_0's binary_logloss: 0.651161
[6]	valid_0's binary_logloss: 0.644638
[7]	valid_0's binary_logloss: 0.642489
[8]	valid_0's binary_logloss: 0.642485
[9]	valid_0's binary_logloss: 0.641966
[10]	valid_0's binary_logloss: 0.640132
[11]	valid_0's binary_logloss: 0.643624
[12]	valid_0's binary_logloss: 0.637071
[13]	valid_0's binary_logloss: 0.642737
[14]	valid_0's binary_logloss: 0.638316
[15]	valid_0's binary_logloss: 0.643204
[16]	valid_0's binary_logloss: 0.645225
[17]	valid_0's binary_logloss: 0.652573
[18]	valid_0's binary_logloss: 0.655003
[19]	valid_0's binary_logloss: 0.653293
[20]	valid_0's binary_logloss: 0.654444
[21]	valid_0's binary_logloss: 0.

[221]	valid_0's binary_logloss: 1.07344
[222]	valid_0's binary_logloss: 1.07468
[223]	valid_0's binary_logloss: 1.07504
[224]	valid_0's binary_logloss: 1.07589
[225]	valid_0's binary_logloss: 1.0791
[226]	valid_0's binary_logloss: 1.08199
[227]	valid_0's binary_logloss: 1.08352
[228]	valid_0's binary_logloss: 1.08711
[229]	valid_0's binary_logloss: 1.08634
[230]	valid_0's binary_logloss: 1.08919
[231]	valid_0's binary_logloss: 1.09242
[232]	valid_0's binary_logloss: 1.09501
[233]	valid_0's binary_logloss: 1.094
[234]	valid_0's binary_logloss: 1.09797
[235]	valid_0's binary_logloss: 1.1005
[236]	valid_0's binary_logloss: 1.1006
[237]	valid_0's binary_logloss: 1.10004
[238]	valid_0's binary_logloss: 1.10193
[239]	valid_0's binary_logloss: 1.10402
[240]	valid_0's binary_logloss: 1.11116
[241]	valid_0's binary_logloss: 1.11329
[242]	valid_0's binary_logloss: 1.11429
[243]	valid_0's binary_logloss: 1.11495
[244]	valid_0's binary_logloss: 1.11625
[245]	valid_0's binary_logloss: 1.12065
[246]

[513]	valid_0's binary_logloss: 1.43166
[514]	valid_0's binary_logloss: 1.43258
[515]	valid_0's binary_logloss: 1.43331
[516]	valid_0's binary_logloss: 1.43241
[517]	valid_0's binary_logloss: 1.43134
[518]	valid_0's binary_logloss: 1.43373
[519]	valid_0's binary_logloss: 1.43653
[520]	valid_0's binary_logloss: 1.4367
[521]	valid_0's binary_logloss: 1.43751
[522]	valid_0's binary_logloss: 1.43745
[523]	valid_0's binary_logloss: 1.43821
[524]	valid_0's binary_logloss: 1.43621
[525]	valid_0's binary_logloss: 1.43711
[526]	valid_0's binary_logloss: 1.43704
[527]	valid_0's binary_logloss: 1.43711
[528]	valid_0's binary_logloss: 1.43987
[529]	valid_0's binary_logloss: 1.44019
[530]	valid_0's binary_logloss: 1.44023
[531]	valid_0's binary_logloss: 1.44058
[532]	valid_0's binary_logloss: 1.44208
[533]	valid_0's binary_logloss: 1.44195
[534]	valid_0's binary_logloss: 1.44209
[535]	valid_0's binary_logloss: 1.44255
[536]	valid_0's binary_logloss: 1.4421
[537]	valid_0's binary_logloss: 1.44237
[5

[719]	valid_0's binary_logloss: 1.55324
[720]	valid_0's binary_logloss: 1.55308
[721]	valid_0's binary_logloss: 1.55286
[722]	valid_0's binary_logloss: 1.55341
[723]	valid_0's binary_logloss: 1.55348
[724]	valid_0's binary_logloss: 1.55273
[725]	valid_0's binary_logloss: 1.55429
[726]	valid_0's binary_logloss: 1.55478
[727]	valid_0's binary_logloss: 1.55507
[728]	valid_0's binary_logloss: 1.55666
[729]	valid_0's binary_logloss: 1.55716
[730]	valid_0's binary_logloss: 1.55846
[731]	valid_0's binary_logloss: 1.55932
[732]	valid_0's binary_logloss: 1.55967
[733]	valid_0's binary_logloss: 1.55971
[734]	valid_0's binary_logloss: 1.55994
[735]	valid_0's binary_logloss: 1.56049
[736]	valid_0's binary_logloss: 1.56064
[737]	valid_0's binary_logloss: 1.56082
[738]	valid_0's binary_logloss: 1.56065
[739]	valid_0's binary_logloss: 1.56091
[740]	valid_0's binary_logloss: 1.5611
[741]	valid_0's binary_logloss: 1.56053
[742]	valid_0's binary_logloss: 1.56075
[743]	valid_0's binary_logloss: 1.56083
[

[234]	valid_0's binary_logloss: 1.25006
[235]	valid_0's binary_logloss: 1.25331
[236]	valid_0's binary_logloss: 1.25288
[237]	valid_0's binary_logloss: 1.25448
[238]	valid_0's binary_logloss: 1.25669
[239]	valid_0's binary_logloss: 1.25882
[240]	valid_0's binary_logloss: 1.26478
[241]	valid_0's binary_logloss: 1.26218
[242]	valid_0's binary_logloss: 1.26038
[243]	valid_0's binary_logloss: 1.26204
[244]	valid_0's binary_logloss: 1.26188
[245]	valid_0's binary_logloss: 1.26249
[246]	valid_0's binary_logloss: 1.26371
[247]	valid_0's binary_logloss: 1.26232
[248]	valid_0's binary_logloss: 1.26333
[249]	valid_0's binary_logloss: 1.26479
[250]	valid_0's binary_logloss: 1.26588
[251]	valid_0's binary_logloss: 1.26727
[252]	valid_0's binary_logloss: 1.26857
[253]	valid_0's binary_logloss: 1.26977
[254]	valid_0's binary_logloss: 1.2688
[255]	valid_0's binary_logloss: 1.27215
[256]	valid_0's binary_logloss: 1.27384
[257]	valid_0's binary_logloss: 1.2742
[258]	valid_0's binary_logloss: 1.28113
[2

[484]	valid_0's binary_logloss: 1.53593
[485]	valid_0's binary_logloss: 1.53636
[486]	valid_0's binary_logloss: 1.53663
[487]	valid_0's binary_logloss: 1.53987
[488]	valid_0's binary_logloss: 1.54123
[489]	valid_0's binary_logloss: 1.54317
[490]	valid_0's binary_logloss: 1.54346
[491]	valid_0's binary_logloss: 1.54499
[492]	valid_0's binary_logloss: 1.54617
[493]	valid_0's binary_logloss: 1.54728
[494]	valid_0's binary_logloss: 1.54861
[495]	valid_0's binary_logloss: 1.54884
[496]	valid_0's binary_logloss: 1.54993
[497]	valid_0's binary_logloss: 1.55097
[498]	valid_0's binary_logloss: 1.552
[499]	valid_0's binary_logloss: 1.55288
[500]	valid_0's binary_logloss: 1.55374
[501]	valid_0's binary_logloss: 1.55431
[502]	valid_0's binary_logloss: 1.55632
[503]	valid_0's binary_logloss: 1.55722
[504]	valid_0's binary_logloss: 1.5574
[505]	valid_0's binary_logloss: 1.56291
[506]	valid_0's binary_logloss: 1.56242
[507]	valid_0's binary_logloss: 1.55956
[508]	valid_0's binary_logloss: 1.56462
[50

[712]	valid_0's binary_logloss: 1.65903
[713]	valid_0's binary_logloss: 1.65834
[714]	valid_0's binary_logloss: 1.65872
[715]	valid_0's binary_logloss: 1.6593
[716]	valid_0's binary_logloss: 1.65944
[717]	valid_0's binary_logloss: 1.66031
[718]	valid_0's binary_logloss: 1.66023
[719]	valid_0's binary_logloss: 1.66052
[720]	valid_0's binary_logloss: 1.65994
[721]	valid_0's binary_logloss: 1.6601
[722]	valid_0's binary_logloss: 1.66088
[723]	valid_0's binary_logloss: 1.66108
[724]	valid_0's binary_logloss: 1.66103
[725]	valid_0's binary_logloss: 1.66115
[726]	valid_0's binary_logloss: 1.66288
[727]	valid_0's binary_logloss: 1.66319
[728]	valid_0's binary_logloss: 1.66363
[729]	valid_0's binary_logloss: 1.66344
[730]	valid_0's binary_logloss: 1.66345
[731]	valid_0's binary_logloss: 1.66436
[732]	valid_0's binary_logloss: 1.66449
[733]	valid_0's binary_logloss: 1.66451
[734]	valid_0's binary_logloss: 1.66476
[735]	valid_0's binary_logloss: 1.66476
[736]	valid_0's binary_logloss: 1.66521
[7

[921]	valid_0's binary_logloss: 1.71599
[922]	valid_0's binary_logloss: 1.716
[923]	valid_0's binary_logloss: 1.71608
[924]	valid_0's binary_logloss: 1.71621
[925]	valid_0's binary_logloss: 1.71624
[926]	valid_0's binary_logloss: 1.71617
[927]	valid_0's binary_logloss: 1.7163
[928]	valid_0's binary_logloss: 1.71663
[929]	valid_0's binary_logloss: 1.71658
[930]	valid_0's binary_logloss: 1.71678
[931]	valid_0's binary_logloss: 1.71686
[932]	valid_0's binary_logloss: 1.71714
[933]	valid_0's binary_logloss: 1.71744
[934]	valid_0's binary_logloss: 1.71754
[935]	valid_0's binary_logloss: 1.71771
[936]	valid_0's binary_logloss: 1.71824
[937]	valid_0's binary_logloss: 1.71823
[938]	valid_0's binary_logloss: 1.71831
[939]	valid_0's binary_logloss: 1.7185
[940]	valid_0's binary_logloss: 1.71857
[941]	valid_0's binary_logloss: 1.7188
[942]	valid_0's binary_logloss: 1.71901
[943]	valid_0's binary_logloss: 1.71975
[944]	valid_0's binary_logloss: 1.71991
[945]	valid_0's binary_logloss: 1.72004
[946]

[214]	valid_0's binary_logloss: 0.878494
[215]	valid_0's binary_logloss: 0.879791
[216]	valid_0's binary_logloss: 0.88137
[217]	valid_0's binary_logloss: 0.881224
[218]	valid_0's binary_logloss: 0.885693
[219]	valid_0's binary_logloss: 0.887821
[220]	valid_0's binary_logloss: 0.888378
[221]	valid_0's binary_logloss: 0.89033
[222]	valid_0's binary_logloss: 0.891438
[223]	valid_0's binary_logloss: 0.889318
[224]	valid_0's binary_logloss: 0.889699
[225]	valid_0's binary_logloss: 0.888534
[226]	valid_0's binary_logloss: 0.88901
[227]	valid_0's binary_logloss: 0.892138
[228]	valid_0's binary_logloss: 0.890129
[229]	valid_0's binary_logloss: 0.893721
[230]	valid_0's binary_logloss: 0.894953
[231]	valid_0's binary_logloss: 0.897482
[232]	valid_0's binary_logloss: 0.901149
[233]	valid_0's binary_logloss: 0.902605
[234]	valid_0's binary_logloss: 0.903115
[235]	valid_0's binary_logloss: 0.903055
[236]	valid_0's binary_logloss: 0.90507
[237]	valid_0's binary_logloss: 0.907307
[238]	valid_0's bina

[484]	valid_0's binary_logloss: 1.24603
[485]	valid_0's binary_logloss: 1.24573
[486]	valid_0's binary_logloss: 1.24605
[487]	valid_0's binary_logloss: 1.2491
[488]	valid_0's binary_logloss: 1.25037
[489]	valid_0's binary_logloss: 1.2513
[490]	valid_0's binary_logloss: 1.25178
[491]	valid_0's binary_logloss: 1.25252
[492]	valid_0's binary_logloss: 1.25425
[493]	valid_0's binary_logloss: 1.25631
[494]	valid_0's binary_logloss: 1.25701
[495]	valid_0's binary_logloss: 1.25763
[496]	valid_0's binary_logloss: 1.25952
[497]	valid_0's binary_logloss: 1.26029
[498]	valid_0's binary_logloss: 1.26099
[499]	valid_0's binary_logloss: 1.263
[500]	valid_0's binary_logloss: 1.26268
[501]	valid_0's binary_logloss: 1.26091
[502]	valid_0's binary_logloss: 1.26113
[503]	valid_0's binary_logloss: 1.26094
[504]	valid_0's binary_logloss: 1.2611
[505]	valid_0's binary_logloss: 1.26445
[506]	valid_0's binary_logloss: 1.26369
[507]	valid_0's binary_logloss: 1.26414
[508]	valid_0's binary_logloss: 1.26465
[509]

[8]	valid_0's binary_logloss: 0.652018
[9]	valid_0's binary_logloss: 0.65017
[10]	valid_0's binary_logloss: 0.654145
[11]	valid_0's binary_logloss: 0.662435
[12]	valid_0's binary_logloss: 0.664625
[13]	valid_0's binary_logloss: 0.675343
[14]	valid_0's binary_logloss: 0.680934
[15]	valid_0's binary_logloss: 0.686321
[16]	valid_0's binary_logloss: 0.681658
[17]	valid_0's binary_logloss: 0.688923
[18]	valid_0's binary_logloss: 0.688066
[19]	valid_0's binary_logloss: 0.688051
[20]	valid_0's binary_logloss: 0.68487
[21]	valid_0's binary_logloss: 0.687681
[22]	valid_0's binary_logloss: 0.692236
[23]	valid_0's binary_logloss: 0.695395
[24]	valid_0's binary_logloss: 0.695025
[25]	valid_0's binary_logloss: 0.695623
[26]	valid_0's binary_logloss: 0.700149
[27]	valid_0's binary_logloss: 0.708112
[28]	valid_0's binary_logloss: 0.703303
[29]	valid_0's binary_logloss: 0.707715
[30]	valid_0's binary_logloss: 0.709533
[31]	valid_0's binary_logloss: 0.717705
[32]	valid_0's binary_logloss: 0.721505
[33]

[293]	valid_0's binary_logloss: 1.42657
[294]	valid_0's binary_logloss: 1.4278
[295]	valid_0's binary_logloss: 1.42631
[296]	valid_0's binary_logloss: 1.42621
[297]	valid_0's binary_logloss: 1.42747
[298]	valid_0's binary_logloss: 1.42975
[299]	valid_0's binary_logloss: 1.43114
[300]	valid_0's binary_logloss: 1.43473
[301]	valid_0's binary_logloss: 1.43659
[302]	valid_0's binary_logloss: 1.43683
[303]	valid_0's binary_logloss: 1.43748
[304]	valid_0's binary_logloss: 1.44033
[305]	valid_0's binary_logloss: 1.44171
[306]	valid_0's binary_logloss: 1.44154
[307]	valid_0's binary_logloss: 1.44503
[308]	valid_0's binary_logloss: 1.4466
[309]	valid_0's binary_logloss: 1.44825
[310]	valid_0's binary_logloss: 1.44928
[311]	valid_0's binary_logloss: 1.44912
[312]	valid_0's binary_logloss: 1.44879
[313]	valid_0's binary_logloss: 1.45113
[314]	valid_0's binary_logloss: 1.45353
[315]	valid_0's binary_logloss: 1.45257
[316]	valid_0's binary_logloss: 1.45437
[317]	valid_0's binary_logloss: 1.45713
[3

[525]	valid_0's binary_logloss: 1.60128
[526]	valid_0's binary_logloss: 1.60126
[527]	valid_0's binary_logloss: 1.60155
[528]	valid_0's binary_logloss: 1.60186
[529]	valid_0's binary_logloss: 1.60259
[530]	valid_0's binary_logloss: 1.60284
[531]	valid_0's binary_logloss: 1.60352
[532]	valid_0's binary_logloss: 1.60373
[533]	valid_0's binary_logloss: 1.60417
[534]	valid_0's binary_logloss: 1.60465
[535]	valid_0's binary_logloss: 1.60491
[536]	valid_0's binary_logloss: 1.60506
[537]	valid_0's binary_logloss: 1.60508
[538]	valid_0's binary_logloss: 1.60522
[539]	valid_0's binary_logloss: 1.60777
[540]	valid_0's binary_logloss: 1.60796
[541]	valid_0's binary_logloss: 1.60791
[542]	valid_0's binary_logloss: 1.60831
[543]	valid_0's binary_logloss: 1.60857
[544]	valid_0's binary_logloss: 1.60855
[545]	valid_0's binary_logloss: 1.60882
[546]	valid_0's binary_logloss: 1.6091
[547]	valid_0's binary_logloss: 1.60921
[548]	valid_0's binary_logloss: 1.60933
[549]	valid_0's binary_logloss: 1.60964
[

[789]	valid_0's binary_logloss: 1.67514
[790]	valid_0's binary_logloss: 1.67518
[791]	valid_0's binary_logloss: 1.67515
[792]	valid_0's binary_logloss: 1.67539
[793]	valid_0's binary_logloss: 1.67553
[794]	valid_0's binary_logloss: 1.67572
[795]	valid_0's binary_logloss: 1.6757
[796]	valid_0's binary_logloss: 1.67571
[797]	valid_0's binary_logloss: 1.67576
[798]	valid_0's binary_logloss: 1.67601
[799]	valid_0's binary_logloss: 1.67631
[800]	valid_0's binary_logloss: 1.67652
[801]	valid_0's binary_logloss: 1.6766
[802]	valid_0's binary_logloss: 1.67676
[803]	valid_0's binary_logloss: 1.67687
[804]	valid_0's binary_logloss: 1.67718
[805]	valid_0's binary_logloss: 1.67743
[806]	valid_0's binary_logloss: 1.67756
[807]	valid_0's binary_logloss: 1.67764
[808]	valid_0's binary_logloss: 1.67784
[809]	valid_0's binary_logloss: 1.67805
[810]	valid_0's binary_logloss: 1.67817
[811]	valid_0's binary_logloss: 1.67819
[812]	valid_0's binary_logloss: 1.6783
[813]	valid_0's binary_logloss: 1.67825
[81

[143]	valid_0's binary_logloss: 1.15173
[144]	valid_0's binary_logloss: 1.15464
[145]	valid_0's binary_logloss: 1.15872
[146]	valid_0's binary_logloss: 1.163
[147]	valid_0's binary_logloss: 1.16669
[148]	valid_0's binary_logloss: 1.17627
[149]	valid_0's binary_logloss: 1.18054
[150]	valid_0's binary_logloss: 1.18499
[151]	valid_0's binary_logloss: 1.19212
[152]	valid_0's binary_logloss: 1.19822
[153]	valid_0's binary_logloss: 1.19802
[154]	valid_0's binary_logloss: 1.20623
[155]	valid_0's binary_logloss: 1.20823
[156]	valid_0's binary_logloss: 1.21193
[157]	valid_0's binary_logloss: 1.21341
[158]	valid_0's binary_logloss: 1.21531
[159]	valid_0's binary_logloss: 1.21969
[160]	valid_0's binary_logloss: 1.22243
[161]	valid_0's binary_logloss: 1.22239
[162]	valid_0's binary_logloss: 1.22355
[163]	valid_0's binary_logloss: 1.22512
[164]	valid_0's binary_logloss: 1.23157
[165]	valid_0's binary_logloss: 1.23392
[166]	valid_0's binary_logloss: 1.23333
[167]	valid_0's binary_logloss: 1.23862
[1

[366]	valid_0's binary_logloss: 0.928536
[367]	valid_0's binary_logloss: 0.92784
[368]	valid_0's binary_logloss: 0.92745
[369]	valid_0's binary_logloss: 0.928059
[370]	valid_0's binary_logloss: 0.927812
[371]	valid_0's binary_logloss: 0.929585
[372]	valid_0's binary_logloss: 0.930512
[373]	valid_0's binary_logloss: 0.932292
[374]	valid_0's binary_logloss: 0.933424
[375]	valid_0's binary_logloss: 0.933471
[376]	valid_0's binary_logloss: 0.93679
[377]	valid_0's binary_logloss: 0.935745
[378]	valid_0's binary_logloss: 0.935544
[379]	valid_0's binary_logloss: 0.935874
[380]	valid_0's binary_logloss: 0.938455
[381]	valid_0's binary_logloss: 0.938964
[382]	valid_0's binary_logloss: 0.939887
[383]	valid_0's binary_logloss: 0.941567
[384]	valid_0's binary_logloss: 0.943822
[385]	valid_0's binary_logloss: 0.945479
[386]	valid_0's binary_logloss: 0.946081
[387]	valid_0's binary_logloss: 0.945917
[388]	valid_0's binary_logloss: 0.948015
[389]	valid_0's binary_logloss: 0.948243
[390]	valid_0's bin

[726]	valid_0's binary_logloss: 1.23692
[727]	valid_0's binary_logloss: 1.23779
[728]	valid_0's binary_logloss: 1.23775
[729]	valid_0's binary_logloss: 1.23939
[730]	valid_0's binary_logloss: 1.23954
[731]	valid_0's binary_logloss: 1.24209
[732]	valid_0's binary_logloss: 1.2424
[733]	valid_0's binary_logloss: 1.2433
[734]	valid_0's binary_logloss: 1.24301
[735]	valid_0's binary_logloss: 1.24386
[736]	valid_0's binary_logloss: 1.24598
[737]	valid_0's binary_logloss: 1.24702
[738]	valid_0's binary_logloss: 1.24829
[739]	valid_0's binary_logloss: 1.24917
[740]	valid_0's binary_logloss: 1.24837
[741]	valid_0's binary_logloss: 1.24763
[742]	valid_0's binary_logloss: 1.24745
[743]	valid_0's binary_logloss: 1.2477
[744]	valid_0's binary_logloss: 1.24781
[745]	valid_0's binary_logloss: 1.24744
[746]	valid_0's binary_logloss: 1.24984
[747]	valid_0's binary_logloss: 1.24991
[748]	valid_0's binary_logloss: 1.24911
[749]	valid_0's binary_logloss: 1.24915
[750]	valid_0's binary_logloss: 1.24996
[75

[281]	valid_0's binary_logloss: 0.639777
[282]	valid_0's binary_logloss: 0.639728
[283]	valid_0's binary_logloss: 0.639717
[284]	valid_0's binary_logloss: 0.639642
[285]	valid_0's binary_logloss: 0.639569
[286]	valid_0's binary_logloss: 0.639536
[287]	valid_0's binary_logloss: 0.639465
[288]	valid_0's binary_logloss: 0.639402
[289]	valid_0's binary_logloss: 0.639332
[290]	valid_0's binary_logloss: 0.639242
[291]	valid_0's binary_logloss: 0.639136
[292]	valid_0's binary_logloss: 0.639069
[293]	valid_0's binary_logloss: 0.638913
[294]	valid_0's binary_logloss: 0.63888
[295]	valid_0's binary_logloss: 0.6388
[296]	valid_0's binary_logloss: 0.638791
[297]	valid_0's binary_logloss: 0.638735
[298]	valid_0's binary_logloss: 0.638694
[299]	valid_0's binary_logloss: 0.63864
[300]	valid_0's binary_logloss: 0.638559
[301]	valid_0's binary_logloss: 0.638507
[302]	valid_0's binary_logloss: 0.638468
[303]	valid_0's binary_logloss: 0.63839
[304]	valid_0's binary_logloss: 0.638343
[305]	valid_0's binar

[336]	valid_0's binary_logloss: 0.685263
[337]	valid_0's binary_logloss: 0.685825
[338]	valid_0's binary_logloss: 0.686549
[339]	valid_0's binary_logloss: 0.68716
[340]	valid_0's binary_logloss: 0.687301
[341]	valid_0's binary_logloss: 0.68769
[342]	valid_0's binary_logloss: 0.687421
[343]	valid_0's binary_logloss: 0.687839
[344]	valid_0's binary_logloss: 0.687501
[345]	valid_0's binary_logloss: 0.688676
[346]	valid_0's binary_logloss: 0.688349
[347]	valid_0's binary_logloss: 0.688553
[348]	valid_0's binary_logloss: 0.689688
[349]	valid_0's binary_logloss: 0.689513
[350]	valid_0's binary_logloss: 0.689343
[351]	valid_0's binary_logloss: 0.689452
[352]	valid_0's binary_logloss: 0.690773
[353]	valid_0's binary_logloss: 0.691684
[354]	valid_0's binary_logloss: 0.692997
[355]	valid_0's binary_logloss: 0.693195
[356]	valid_0's binary_logloss: 0.694498
[357]	valid_0's binary_logloss: 0.695162
[358]	valid_0's binary_logloss: 0.694221
[359]	valid_0's binary_logloss: 0.694619
[360]	valid_0's bi

[704]	valid_0's binary_logloss: 0.823329
[705]	valid_0's binary_logloss: 0.82339
[706]	valid_0's binary_logloss: 0.823554
[707]	valid_0's binary_logloss: 0.823661
[708]	valid_0's binary_logloss: 0.823291
[709]	valid_0's binary_logloss: 0.823935
[710]	valid_0's binary_logloss: 0.825591
[711]	valid_0's binary_logloss: 0.826165
[712]	valid_0's binary_logloss: 0.826851
[713]	valid_0's binary_logloss: 0.82744
[714]	valid_0's binary_logloss: 0.828347
[715]	valid_0's binary_logloss: 0.82865
[716]	valid_0's binary_logloss: 0.828498
[717]	valid_0's binary_logloss: 0.828984
[718]	valid_0's binary_logloss: 0.828763
[719]	valid_0's binary_logloss: 0.828885
[720]	valid_0's binary_logloss: 0.829457
[721]	valid_0's binary_logloss: 0.830376
[722]	valid_0's binary_logloss: 0.830305
[723]	valid_0's binary_logloss: 0.830153
[724]	valid_0's binary_logloss: 0.830669
[725]	valid_0's binary_logloss: 0.829389
[726]	valid_0's binary_logloss: 0.831733
[727]	valid_0's binary_logloss: 0.832026
[728]	valid_0's bin

[2]	valid_0's binary_logloss: 0.64908
[3]	valid_0's binary_logloss: 0.64292
[4]	valid_0's binary_logloss: 0.635871
[5]	valid_0's binary_logloss: 0.640143
[6]	valid_0's binary_logloss: 0.642542
[7]	valid_0's binary_logloss: 0.638694
[8]	valid_0's binary_logloss: 0.640333
[9]	valid_0's binary_logloss: 0.642574
[10]	valid_0's binary_logloss: 0.641173
[11]	valid_0's binary_logloss: 0.646504
[12]	valid_0's binary_logloss: 0.652199
[13]	valid_0's binary_logloss: 0.656853
[14]	valid_0's binary_logloss: 0.667355
[15]	valid_0's binary_logloss: 0.663181
[16]	valid_0's binary_logloss: 0.672851
[17]	valid_0's binary_logloss: 0.670582
[18]	valid_0's binary_logloss: 0.675915
[19]	valid_0's binary_logloss: 0.675473
[20]	valid_0's binary_logloss: 0.683488
[21]	valid_0's binary_logloss: 0.687519
[22]	valid_0's binary_logloss: 0.69358
[23]	valid_0's binary_logloss: 0.697865
[24]	valid_0's binary_logloss: 0.699498
[25]	valid_0's binary_logloss: 0.697268
[26]	valid_0's binary_logloss: 0.705301
[27]	valid_

[336]	valid_0's binary_logloss: 0.969819
[337]	valid_0's binary_logloss: 0.970774
[338]	valid_0's binary_logloss: 0.970179
[339]	valid_0's binary_logloss: 0.970827
[340]	valid_0's binary_logloss: 0.971555
[341]	valid_0's binary_logloss: 0.970091
[342]	valid_0's binary_logloss: 0.970875
[343]	valid_0's binary_logloss: 0.974986
[344]	valid_0's binary_logloss: 0.975569
[345]	valid_0's binary_logloss: 0.975347
[346]	valid_0's binary_logloss: 0.976904
[347]	valid_0's binary_logloss: 0.977839
[348]	valid_0's binary_logloss: 0.979245
[349]	valid_0's binary_logloss: 0.980029
[350]	valid_0's binary_logloss: 0.983818
[351]	valid_0's binary_logloss: 0.984839
[352]	valid_0's binary_logloss: 0.986334
[353]	valid_0's binary_logloss: 0.989696
[354]	valid_0's binary_logloss: 0.991434
[355]	valid_0's binary_logloss: 0.995428
[356]	valid_0's binary_logloss: 0.995588
[357]	valid_0's binary_logloss: 0.995844
[358]	valid_0's binary_logloss: 0.998373
[359]	valid_0's binary_logloss: 0.998794
[360]	valid_0's 

[663]	valid_0's binary_logloss: 1.29074
[664]	valid_0's binary_logloss: 1.29189
[665]	valid_0's binary_logloss: 1.29191
[666]	valid_0's binary_logloss: 1.29604
[667]	valid_0's binary_logloss: 1.29668
[668]	valid_0's binary_logloss: 1.29777
[669]	valid_0's binary_logloss: 1.29958
[670]	valid_0's binary_logloss: 1.29922
[671]	valid_0's binary_logloss: 1.29873
[672]	valid_0's binary_logloss: 1.29829
[673]	valid_0's binary_logloss: 1.30279
[674]	valid_0's binary_logloss: 1.30444
[675]	valid_0's binary_logloss: 1.30518
[676]	valid_0's binary_logloss: 1.30619
[677]	valid_0's binary_logloss: 1.30821
[678]	valid_0's binary_logloss: 1.30942
[679]	valid_0's binary_logloss: 1.30756
[680]	valid_0's binary_logloss: 1.30668
[681]	valid_0's binary_logloss: 1.3068
[682]	valid_0's binary_logloss: 1.30667
[683]	valid_0's binary_logloss: 1.30685
[684]	valid_0's binary_logloss: 1.30685
[685]	valid_0's binary_logloss: 1.3067
[686]	valid_0's binary_logloss: 1.30708
[687]	valid_0's binary_logloss: 1.30581
[6

[108]	valid_0's binary_logloss: 0.638617
[109]	valid_0's binary_logloss: 0.638533
[110]	valid_0's binary_logloss: 0.638845
[111]	valid_0's binary_logloss: 0.639322
[112]	valid_0's binary_logloss: 0.639876
[113]	valid_0's binary_logloss: 0.639727
[114]	valid_0's binary_logloss: 0.639598
[115]	valid_0's binary_logloss: 0.639755
[116]	valid_0's binary_logloss: 0.639389
[117]	valid_0's binary_logloss: 0.639678
[118]	valid_0's binary_logloss: 0.639807
[119]	valid_0's binary_logloss: 0.639942
[120]	valid_0's binary_logloss: 0.640546
[121]	valid_0's binary_logloss: 0.640587
[122]	valid_0's binary_logloss: 0.640593
[123]	valid_0's binary_logloss: 0.640758
[124]	valid_0's binary_logloss: 0.640611
[125]	valid_0's binary_logloss: 0.640681
[126]	valid_0's binary_logloss: 0.640856
[127]	valid_0's binary_logloss: 0.640733
[128]	valid_0's binary_logloss: 0.641143
[129]	valid_0's binary_logloss: 0.641516
[130]	valid_0's binary_logloss: 0.641528
[131]	valid_0's binary_logloss: 0.641892
[132]	valid_0's 

[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.683889
[2]	valid_0's binary_logloss: 0.675372
[3]	valid_0's binary_logloss: 0.668257
[4]	valid_0's binary_logloss: 0.661954
[5]	valid_0's binary_logloss: 0.657785
[6]	valid_0's binary_logloss: 0.653669
[7]	valid_0's binary_logloss: 0.649768
[8]	valid_0's binary_logloss: 0.645701
[9]	valid_0's binary_logloss: 0.64396
[10]	valid_0's binary_logloss: 0.640712
[11]	valid_0's binary_logloss: 0.637364
[12]	valid_0's binary_logloss: 0.634522
[13]	valid_0's binary_logloss: 0.632659
[14]	valid_0's binary_logloss: 0.631449
[15]	valid_0's binary_logloss: 0.631298
[16]	valid_0's binary_logloss: 0.629172
[17]	valid_0's binary_logloss: 0.628611
[18]	valid_0's binary_logloss: 0.627806
[19]	valid_0's binary_logloss: 0.629444
[20]	valid_0's binary_logloss: 0.628096
[21]	valid_0's binary_logloss: 0.627048
[22]	valid_0's binary_logloss: 0.628728
[23]	valid_0's binary_logloss: 0.630504
[24]	valid_0's binary_logloss: 0.62997

[385]	valid_0's binary_logloss: 0.859667
[386]	valid_0's binary_logloss: 0.859388
[387]	valid_0's binary_logloss: 0.859454
[388]	valid_0's binary_logloss: 0.859412
[389]	valid_0's binary_logloss: 0.8612
[390]	valid_0's binary_logloss: 0.862305
[391]	valid_0's binary_logloss: 0.863375
[392]	valid_0's binary_logloss: 0.86636
[393]	valid_0's binary_logloss: 0.866773
[394]	valid_0's binary_logloss: 0.867024
[395]	valid_0's binary_logloss: 0.867652
[396]	valid_0's binary_logloss: 0.86763
[397]	valid_0's binary_logloss: 0.868487
[398]	valid_0's binary_logloss: 0.870031
[399]	valid_0's binary_logloss: 0.871184
[400]	valid_0's binary_logloss: 0.872253
[401]	valid_0's binary_logloss: 0.871681
[402]	valid_0's binary_logloss: 0.873576
[403]	valid_0's binary_logloss: 0.873712
[404]	valid_0's binary_logloss: 0.875131
[405]	valid_0's binary_logloss: 0.876886
[406]	valid_0's binary_logloss: 0.877827
[407]	valid_0's binary_logloss: 0.878402
[408]	valid_0's binary_logloss: 0.878436
[409]	valid_0's bina

[612]	valid_0's binary_logloss: 1.01983
[613]	valid_0's binary_logloss: 1.02172
[614]	valid_0's binary_logloss: 1.02278
[615]	valid_0's binary_logloss: 1.02286
[616]	valid_0's binary_logloss: 1.02388
[617]	valid_0's binary_logloss: 1.02445
[618]	valid_0's binary_logloss: 1.02503
[619]	valid_0's binary_logloss: 1.02454
[620]	valid_0's binary_logloss: 1.0241
[621]	valid_0's binary_logloss: 1.02523
[622]	valid_0's binary_logloss: 1.02563
[623]	valid_0's binary_logloss: 1.02478
[624]	valid_0's binary_logloss: 1.0247
[625]	valid_0's binary_logloss: 1.02581
[626]	valid_0's binary_logloss: 1.02695
[627]	valid_0's binary_logloss: 1.02747
[628]	valid_0's binary_logloss: 1.02912
[629]	valid_0's binary_logloss: 1.03038
[630]	valid_0's binary_logloss: 1.03165
[631]	valid_0's binary_logloss: 1.03242
[632]	valid_0's binary_logloss: 1.03317
[633]	valid_0's binary_logloss: 1.03453
[634]	valid_0's binary_logloss: 1.03358
[635]	valid_0's binary_logloss: 1.03556
[636]	valid_0's binary_logloss: 1.03663
[6

[831]	valid_0's binary_logloss: 1.14036
[832]	valid_0's binary_logloss: 1.14174
[833]	valid_0's binary_logloss: 1.14199
[834]	valid_0's binary_logloss: 1.14317
[835]	valid_0's binary_logloss: 1.1433
[836]	valid_0's binary_logloss: 1.14382
[837]	valid_0's binary_logloss: 1.14602
[838]	valid_0's binary_logloss: 1.14586
[839]	valid_0's binary_logloss: 1.14681
[840]	valid_0's binary_logloss: 1.14712
[841]	valid_0's binary_logloss: 1.14839
[842]	valid_0's binary_logloss: 1.14854
[843]	valid_0's binary_logloss: 1.14888
[844]	valid_0's binary_logloss: 1.15106
[845]	valid_0's binary_logloss: 1.15147
[846]	valid_0's binary_logloss: 1.15217
[847]	valid_0's binary_logloss: 1.15238
[848]	valid_0's binary_logloss: 1.153
[849]	valid_0's binary_logloss: 1.1535
[850]	valid_0's binary_logloss: 1.15336
[851]	valid_0's binary_logloss: 1.15525
[852]	valid_0's binary_logloss: 1.15466
[853]	valid_0's binary_logloss: 1.15467
[854]	valid_0's binary_logloss: 1.15496
[855]	valid_0's binary_logloss: 1.15601
[856

[385]	valid_0's binary_logloss: 0.808412
[386]	valid_0's binary_logloss: 0.808544
[387]	valid_0's binary_logloss: 0.808779
[388]	valid_0's binary_logloss: 0.810107
[389]	valid_0's binary_logloss: 0.811561
[390]	valid_0's binary_logloss: 0.811746
[391]	valid_0's binary_logloss: 0.812221
[392]	valid_0's binary_logloss: 0.81205
[393]	valid_0's binary_logloss: 0.814365
[394]	valid_0's binary_logloss: 0.815608
[395]	valid_0's binary_logloss: 0.816317
[396]	valid_0's binary_logloss: 0.817462
[397]	valid_0's binary_logloss: 0.819832
[398]	valid_0's binary_logloss: 0.820953
[399]	valid_0's binary_logloss: 0.821533
[400]	valid_0's binary_logloss: 0.822173
[401]	valid_0's binary_logloss: 0.822902
[402]	valid_0's binary_logloss: 0.822838
[403]	valid_0's binary_logloss: 0.824452
[404]	valid_0's binary_logloss: 0.825011
[405]	valid_0's binary_logloss: 0.82478
[406]	valid_0's binary_logloss: 0.82629
[407]	valid_0's binary_logloss: 0.827938
[408]	valid_0's binary_logloss: 0.829553
[409]	valid_0's bin

| 15        | 0.6368    | 0.1242    | 3.996     | 697.0     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.674231
[2]	valid_0's binary_logloss: 0.661448
[3]	valid_0's binary_logloss: 0.650079
[4]	valid_0's binary_logloss: 0.642737
[5]	valid_0's binary_logloss: 0.63451
[6]	valid_0's binary_logloss: 0.639856
[7]	valid_0's binary_logloss: 0.640729
[8]	valid_0's binary_logloss: 0.64751
[9]	valid_0's binary_logloss: 0.653682
[10]	valid_0's binary_logloss: 0.65637
[11]	valid_0's binary_logloss: 0.659496
[12]	valid_0's binary_logloss: 0.666501
[13]	valid_0's binary_logloss: 0.671991
[14]	valid_0's binary_logloss: 0.675141
[15]	valid_0's binary_logloss: 0.678488
[16]	valid_0's binary_logloss: 0.685024
[17]	valid_0's binary_logloss: 0.687266
[18]	valid_0's binary_logloss: 0.687337
[19]	valid_0's binary_logloss: 0.690494
[20]	valid_0's binary_logloss: 0.692953
[21]	valid_0's binary_logloss: 0.69305
[22]	valid_0's binary_logloss: 0.696152
[23]	valid_0's bina

[376]	valid_0's binary_logloss: 1.38497
[377]	valid_0's binary_logloss: 1.38569
[378]	valid_0's binary_logloss: 1.38531
[379]	valid_0's binary_logloss: 1.38844
[380]	valid_0's binary_logloss: 1.39129
[381]	valid_0's binary_logloss: 1.39113
[382]	valid_0's binary_logloss: 1.39257
[383]	valid_0's binary_logloss: 1.39209
[384]	valid_0's binary_logloss: 1.39252
[385]	valid_0's binary_logloss: 1.39661
[386]	valid_0's binary_logloss: 1.40155
[387]	valid_0's binary_logloss: 1.40154
[388]	valid_0's binary_logloss: 1.40251
[389]	valid_0's binary_logloss: 1.4031
[390]	valid_0's binary_logloss: 1.40348
[391]	valid_0's binary_logloss: 1.4027
[392]	valid_0's binary_logloss: 1.40406
[393]	valid_0's binary_logloss: 1.40534
[394]	valid_0's binary_logloss: 1.40784
[395]	valid_0's binary_logloss: 1.40995
[396]	valid_0's binary_logloss: 1.41198
[397]	valid_0's binary_logloss: 1.41078
[398]	valid_0's binary_logloss: 1.41351
[399]	valid_0's binary_logloss: 1.41393
[400]	valid_0's binary_logloss: 1.41499
[4

[143]	valid_0's binary_logloss: 0.984531
[144]	valid_0's binary_logloss: 0.983887
[145]	valid_0's binary_logloss: 0.984468
[146]	valid_0's binary_logloss: 0.989285
[147]	valid_0's binary_logloss: 0.989755
[148]	valid_0's binary_logloss: 0.9931
[149]	valid_0's binary_logloss: 1.00058
[150]	valid_0's binary_logloss: 1.00235
[151]	valid_0's binary_logloss: 1.00278
[152]	valid_0's binary_logloss: 1.00577
[153]	valid_0's binary_logloss: 1.01033
[154]	valid_0's binary_logloss: 1.01703
[155]	valid_0's binary_logloss: 1.0197
[156]	valid_0's binary_logloss: 1.02457
[157]	valid_0's binary_logloss: 1.02503
[158]	valid_0's binary_logloss: 1.02528
[159]	valid_0's binary_logloss: 1.03191
[160]	valid_0's binary_logloss: 1.03773
[161]	valid_0's binary_logloss: 1.03766
[162]	valid_0's binary_logloss: 1.04207
[163]	valid_0's binary_logloss: 1.045
[164]	valid_0's binary_logloss: 1.04588
[165]	valid_0's binary_logloss: 1.04948
[166]	valid_0's binary_logloss: 1.0519
[167]	valid_0's binary_logloss: 1.05585


[359]	valid_0's binary_logloss: 1.3594
[360]	valid_0's binary_logloss: 1.36
[361]	valid_0's binary_logloss: 1.36006
[362]	valid_0's binary_logloss: 1.36029
[363]	valid_0's binary_logloss: 1.36579
[364]	valid_0's binary_logloss: 1.36622
[365]	valid_0's binary_logloss: 1.36612
[366]	valid_0's binary_logloss: 1.36587
[367]	valid_0's binary_logloss: 1.36684
[368]	valid_0's binary_logloss: 1.37254
[369]	valid_0's binary_logloss: 1.37351
[370]	valid_0's binary_logloss: 1.37637
[371]	valid_0's binary_logloss: 1.37641
[372]	valid_0's binary_logloss: 1.37976
[373]	valid_0's binary_logloss: 1.37891
[374]	valid_0's binary_logloss: 1.38221
[375]	valid_0's binary_logloss: 1.38346
[376]	valid_0's binary_logloss: 1.38655
[377]	valid_0's binary_logloss: 1.38766
[378]	valid_0's binary_logloss: 1.38789
[379]	valid_0's binary_logloss: 1.38796
[380]	valid_0's binary_logloss: 1.39196
[381]	valid_0's binary_logloss: 1.39301
[382]	valid_0's binary_logloss: 1.39355
[383]	valid_0's binary_logloss: 1.39541
[384

[166]	valid_0's binary_logloss: 0.966261
[167]	valid_0's binary_logloss: 0.965583
[168]	valid_0's binary_logloss: 0.965595
[169]	valid_0's binary_logloss: 0.965427
[170]	valid_0's binary_logloss: 0.970816
[171]	valid_0's binary_logloss: 0.971077
[172]	valid_0's binary_logloss: 0.977382
[173]	valid_0's binary_logloss: 0.981605
[174]	valid_0's binary_logloss: 0.986488
[175]	valid_0's binary_logloss: 0.986331
[176]	valid_0's binary_logloss: 0.99081
[177]	valid_0's binary_logloss: 0.993301
[178]	valid_0's binary_logloss: 0.994454
[179]	valid_0's binary_logloss: 0.997086
[180]	valid_0's binary_logloss: 0.996476
[181]	valid_0's binary_logloss: 0.999423
[182]	valid_0's binary_logloss: 0.999418
[183]	valid_0's binary_logloss: 1.00236
[184]	valid_0's binary_logloss: 1.00572
[185]	valid_0's binary_logloss: 1.00772
[186]	valid_0's binary_logloss: 1.01079
[187]	valid_0's binary_logloss: 1.01461
[188]	valid_0's binary_logloss: 1.01794
[189]	valid_0's binary_logloss: 1.02022
[190]	valid_0's binary_l

| 18        | 0.6404    | 0.1359    | 8.292     | 479.0     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.691069
[2]	valid_0's binary_logloss: 0.688901
[3]	valid_0's binary_logloss: 0.68668
[4]	valid_0's binary_logloss: 0.684665
[5]	valid_0's binary_logloss: 0.682595
[6]	valid_0's binary_logloss: 0.680599
[7]	valid_0's binary_logloss: 0.678793
[8]	valid_0's binary_logloss: 0.677043
[9]	valid_0's binary_logloss: 0.675389
[10]	valid_0's binary_logloss: 0.673769
[11]	valid_0's binary_logloss: 0.672145
[12]	valid_0's binary_logloss: 0.670632
[13]	valid_0's binary_logloss: 0.669066
[14]	valid_0's binary_logloss: 0.667939
[15]	valid_0's binary_logloss: 0.667032
[16]	valid_0's binary_logloss: 0.666173
[17]	valid_0's binary_logloss: 0.665361
[18]	valid_0's binary_logloss: 0.664595
[19]	valid_0's binary_logloss: 0.663872
[20]	valid_0's binary_logloss: 0.663191
[21]	valid_0's binary_logloss: 0.662616
[22]	valid_0's binary_logloss: 0.66199
[23]	valid_0's bi

[1]	valid_0's binary_logloss: 0.675196
[2]	valid_0's binary_logloss: 0.662753
[3]	valid_0's binary_logloss: 0.657047
[4]	valid_0's binary_logloss: 0.653856
[5]	valid_0's binary_logloss: 0.650332
[6]	valid_0's binary_logloss: 0.650033
[7]	valid_0's binary_logloss: 0.64452
[8]	valid_0's binary_logloss: 0.642838
[9]	valid_0's binary_logloss: 0.643729
[10]	valid_0's binary_logloss: 0.641245
[11]	valid_0's binary_logloss: 0.637808
[12]	valid_0's binary_logloss: 0.631957
[13]	valid_0's binary_logloss: 0.629584
[14]	valid_0's binary_logloss: 0.629152
[15]	valid_0's binary_logloss: 0.62828
[16]	valid_0's binary_logloss: 0.629971
[17]	valid_0's binary_logloss: 0.631085
[18]	valid_0's binary_logloss: 0.63113
[19]	valid_0's binary_logloss: 0.632231
[20]	valid_0's binary_logloss: 0.633567
[21]	valid_0's binary_logloss: 0.635921
[22]	valid_0's binary_logloss: 0.633084
[23]	valid_0's binary_logloss: 0.632422
[24]	valid_0's binary_logloss: 0.633561
[25]	valid_0's binary_logloss: 0.634205
[26]	valid_0

[380]	valid_0's binary_logloss: 0.842312
[381]	valid_0's binary_logloss: 0.843646
[382]	valid_0's binary_logloss: 0.844273
[383]	valid_0's binary_logloss: 0.845605
[384]	valid_0's binary_logloss: 0.847336
[385]	valid_0's binary_logloss: 0.848767
[386]	valid_0's binary_logloss: 0.849832
[387]	valid_0's binary_logloss: 0.85003
[388]	valid_0's binary_logloss: 0.850336
[389]	valid_0's binary_logloss: 0.851408
[390]	valid_0's binary_logloss: 0.851696
[391]	valid_0's binary_logloss: 0.851631
[392]	valid_0's binary_logloss: 0.852885
[393]	valid_0's binary_logloss: 0.854199
[394]	valid_0's binary_logloss: 0.854721
[395]	valid_0's binary_logloss: 0.854173
[396]	valid_0's binary_logloss: 0.856433
[397]	valid_0's binary_logloss: 0.856542
[398]	valid_0's binary_logloss: 0.856346
[399]	valid_0's binary_logloss: 0.855701
[400]	valid_0's binary_logloss: 0.855579
[401]	valid_0's binary_logloss: 0.855964
[402]	valid_0's binary_logloss: 0.857756
[403]	valid_0's binary_logloss: 0.858816
[404]	valid_0's b

[730]	valid_0's binary_logloss: 1.0863
[731]	valid_0's binary_logloss: 1.0864
[732]	valid_0's binary_logloss: 1.08723
[733]	valid_0's binary_logloss: 1.08691
[734]	valid_0's binary_logloss: 1.08674
[735]	valid_0's binary_logloss: 1.08661
[736]	valid_0's binary_logloss: 1.08633
[737]	valid_0's binary_logloss: 1.08706
[738]	valid_0's binary_logloss: 1.08873
[739]	valid_0's binary_logloss: 1.08933
[740]	valid_0's binary_logloss: 1.08986
[741]	valid_0's binary_logloss: 1.09005
[742]	valid_0's binary_logloss: 1.08933
[743]	valid_0's binary_logloss: 1.0904
[744]	valid_0's binary_logloss: 1.09095
[745]	valid_0's binary_logloss: 1.09251
[746]	valid_0's binary_logloss: 1.09301
[747]	valid_0's binary_logloss: 1.09186
[748]	valid_0's binary_logloss: 1.09388
[749]	valid_0's binary_logloss: 1.09524
[750]	valid_0's binary_logloss: 1.09563
[751]	valid_0's binary_logloss: 1.09553
[752]	valid_0's binary_logloss: 1.09733
[753]	valid_0's binary_logloss: 1.09822
[754]	valid_0's binary_logloss: 1.09714
[75

[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.681576
[2]	valid_0's binary_logloss: 0.671844
[3]	valid_0's binary_logloss: 0.663701
[4]	valid_0's binary_logloss: 0.65742
[5]	valid_0's binary_logloss: 0.652048
[6]	valid_0's binary_logloss: 0.647429
[7]	valid_0's binary_logloss: 0.645618
[8]	valid_0's binary_logloss: 0.641142
[9]	valid_0's binary_logloss: 0.642421
[10]	valid_0's binary_logloss: 0.638561
[11]	valid_0's binary_logloss: 0.635577
[12]	valid_0's binary_logloss: 0.632439
[13]	valid_0's binary_logloss: 0.630694
[14]	valid_0's binary_logloss: 0.633097
[15]	valid_0's binary_logloss: 0.630799
[16]	valid_0's binary_logloss: 0.626647
[17]	valid_0's binary_logloss: 0.627179
[18]	valid_0's binary_logloss: 0.629008
[19]	valid_0's binary_logloss: 0.628731
[20]	valid_0's binary_logloss: 0.629419
[21]	valid_0's binary_logloss: 0.631277
[22]	valid_0's binary_logloss: 0.633817
[23]	valid_0's binary_logloss: 0.635914
[24]	valid_0's binary_logloss: 0.63510

[386]	valid_0's binary_logloss: 0.969401
[387]	valid_0's binary_logloss: 0.971177
[388]	valid_0's binary_logloss: 0.972459
[389]	valid_0's binary_logloss: 0.973428
[390]	valid_0's binary_logloss: 0.972186
[391]	valid_0's binary_logloss: 0.97343
[392]	valid_0's binary_logloss: 0.972741
[393]	valid_0's binary_logloss: 0.974042
[394]	valid_0's binary_logloss: 0.974357
[395]	valid_0's binary_logloss: 0.974458
[396]	valid_0's binary_logloss: 0.975252
[397]	valid_0's binary_logloss: 0.976057
[398]	valid_0's binary_logloss: 0.977835
[399]	valid_0's binary_logloss: 0.978279
[400]	valid_0's binary_logloss: 0.978103
[401]	valid_0's binary_logloss: 0.977719
[402]	valid_0's binary_logloss: 0.978985
[403]	valid_0's binary_logloss: 0.980324
[404]	valid_0's binary_logloss: 0.98054
[405]	valid_0's binary_logloss: 0.980854
[406]	valid_0's binary_logloss: 0.982405
[407]	valid_0's binary_logloss: 0.984518
[408]	valid_0's binary_logloss: 0.986099
[409]	valid_0's binary_logloss: 0.987026
[410]	valid_0's bi

[639]	valid_0's binary_logloss: 1.16663
[640]	valid_0's binary_logloss: 1.16613
[641]	valid_0's binary_logloss: 1.16647
[642]	valid_0's binary_logloss: 1.16588
[643]	valid_0's binary_logloss: 1.16739
[644]	valid_0's binary_logloss: 1.16784
[645]	valid_0's binary_logloss: 1.16798
[646]	valid_0's binary_logloss: 1.16823
[647]	valid_0's binary_logloss: 1.16925
[648]	valid_0's binary_logloss: 1.17004
[649]	valid_0's binary_logloss: 1.17172
[650]	valid_0's binary_logloss: 1.17502
[651]	valid_0's binary_logloss: 1.17529
[652]	valid_0's binary_logloss: 1.1758
[653]	valid_0's binary_logloss: 1.177
[654]	valid_0's binary_logloss: 1.17795
[655]	valid_0's binary_logloss: 1.17825
[656]	valid_0's binary_logloss: 1.17838
[657]	valid_0's binary_logloss: 1.17877
[658]	valid_0's binary_logloss: 1.17994
[659]	valid_0's binary_logloss: 1.18005
[660]	valid_0's binary_logloss: 1.18191
[661]	valid_0's binary_logloss: 1.18348
[662]	valid_0's binary_logloss: 1.18351
[663]	valid_0's binary_logloss: 1.18471
[66

[184]	valid_0's binary_logloss: 1.09733
[185]	valid_0's binary_logloss: 1.09652
[186]	valid_0's binary_logloss: 1.09806
[187]	valid_0's binary_logloss: 1.10031
[188]	valid_0's binary_logloss: 1.10232
[189]	valid_0's binary_logloss: 1.10381
[190]	valid_0's binary_logloss: 1.10139
[191]	valid_0's binary_logloss: 1.10121
[192]	valid_0's binary_logloss: 1.10172
[193]	valid_0's binary_logloss: 1.10322
[194]	valid_0's binary_logloss: 1.10509
[195]	valid_0's binary_logloss: 1.10902
[196]	valid_0's binary_logloss: 1.11217
[197]	valid_0's binary_logloss: 1.11843
[198]	valid_0's binary_logloss: 1.12295
[199]	valid_0's binary_logloss: 1.12583
[200]	valid_0's binary_logloss: 1.12591
[201]	valid_0's binary_logloss: 1.13029
[202]	valid_0's binary_logloss: 1.13006
[203]	valid_0's binary_logloss: 1.13135
[204]	valid_0's binary_logloss: 1.13745
[205]	valid_0's binary_logloss: 1.13837
[206]	valid_0's binary_logloss: 1.13719
[207]	valid_0's binary_logloss: 1.13931
[208]	valid_0's binary_logloss: 1.14236


[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.682976
[2]	valid_0's binary_logloss: 0.674596
[3]	valid_0's binary_logloss: 0.666594
[4]	valid_0's binary_logloss: 0.660286
[5]	valid_0's binary_logloss: 0.656104
[6]	valid_0's binary_logloss: 0.653141
[7]	valid_0's binary_logloss: 0.651201
[8]	valid_0's binary_logloss: 0.64689
[9]	valid_0's binary_logloss: 0.643692
[10]	valid_0's binary_logloss: 0.644218
[11]	valid_0's binary_logloss: 0.642941
[12]	valid_0's binary_logloss: 0.64142
[13]	valid_0's binary_logloss: 0.640249
[14]	valid_0's binary_logloss: 0.639982
[15]	valid_0's binary_logloss: 0.637314
[16]	valid_0's binary_logloss: 0.639985
[17]	valid_0's binary_logloss: 0.641863
[18]	valid_0's binary_logloss: 0.644398
[19]	valid_0's binary_logloss: 0.644265
[20]	valid_0's binary_logloss: 0.644351
[21]	valid_0's binary_logloss: 0.643539
[22]	valid_0's binary_logloss: 0.642762
[23]	valid_0's binary_logloss: 0.640574
[24]	valid_0's binary_logloss: 0.639997

[67]	valid_0's binary_logloss: 0.646372
[68]	valid_0's binary_logloss: 0.645693
[69]	valid_0's binary_logloss: 0.645379
[70]	valid_0's binary_logloss: 0.647512
[71]	valid_0's binary_logloss: 0.648754
[72]	valid_0's binary_logloss: 0.648501
[73]	valid_0's binary_logloss: 0.649425
[74]	valid_0's binary_logloss: 0.649955
[75]	valid_0's binary_logloss: 0.649417
[76]	valid_0's binary_logloss: 0.651675
[77]	valid_0's binary_logloss: 0.651584
[78]	valid_0's binary_logloss: 0.652771
[79]	valid_0's binary_logloss: 0.654462
[80]	valid_0's binary_logloss: 0.655412
[81]	valid_0's binary_logloss: 0.651943
[82]	valid_0's binary_logloss: 0.65246
[83]	valid_0's binary_logloss: 0.653518
[84]	valid_0's binary_logloss: 0.65564
[85]	valid_0's binary_logloss: 0.653574
[86]	valid_0's binary_logloss: 0.650512
[87]	valid_0's binary_logloss: 0.652479
[88]	valid_0's binary_logloss: 0.655305
[89]	valid_0's binary_logloss: 0.655716
[90]	valid_0's binary_logloss: 0.658938
[91]	valid_0's binary_logloss: 0.657493
[9

| 25        | 0.67      | 0.2389    | 2.998     | 528.2     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.692207
[2]	valid_0's binary_logloss: 0.69065
[3]	valid_0's binary_logloss: 0.689477
[4]	valid_0's binary_logloss: 0.688123
[5]	valid_0's binary_logloss: 0.686705
[6]	valid_0's binary_logloss: 0.685621
[7]	valid_0's binary_logloss: 0.684469
[8]	valid_0's binary_logloss: 0.683582
[9]	valid_0's binary_logloss: 0.682625
[10]	valid_0's binary_logloss: 0.681799
[11]	valid_0's binary_logloss: 0.680571
[12]	valid_0's binary_logloss: 0.679623
[13]	valid_0's binary_logloss: 0.67876
[14]	valid_0's binary_logloss: 0.67791
[15]	valid_0's binary_logloss: 0.676794
[16]	valid_0's binary_logloss: 0.675953
[17]	valid_0's binary_logloss: 0.675176
[18]	valid_0's binary_logloss: 0.674156
[19]	valid_0's binary_logloss: 0.673281
[20]	valid_0's binary_logloss: 0.672444
[21]	valid_0's binary_logloss: 0.671798
[22]	valid_0's binary_logloss: 0.670995
[23]	valid_0's bin

[273]	valid_0's binary_logloss: 0.654195
[274]	valid_0's binary_logloss: 0.654535
[275]	valid_0's binary_logloss: 0.654903
[276]	valid_0's binary_logloss: 0.654897
[277]	valid_0's binary_logloss: 0.655125
[278]	valid_0's binary_logloss: 0.655533
[279]	valid_0's binary_logloss: 0.655709
[280]	valid_0's binary_logloss: 0.655834
[281]	valid_0's binary_logloss: 0.65577
[282]	valid_0's binary_logloss: 0.655864
[283]	valid_0's binary_logloss: 0.65614
[284]	valid_0's binary_logloss: 0.656082
[285]	valid_0's binary_logloss: 0.655718
[286]	valid_0's binary_logloss: 0.656034
[287]	valid_0's binary_logloss: 0.656172
[288]	valid_0's binary_logloss: 0.656098
[289]	valid_0's binary_logloss: 0.655982
[290]	valid_0's binary_logloss: 0.656071
[291]	valid_0's binary_logloss: 0.656363
[292]	valid_0's binary_logloss: 0.656351
[293]	valid_0's binary_logloss: 0.656333
[294]	valid_0's binary_logloss: 0.656325
[295]	valid_0's binary_logloss: 0.656558
[296]	valid_0's binary_logloss: 0.656879
[297]	valid_0's bi

[567]	valid_0's binary_logloss: 0.694078
[568]	valid_0's binary_logloss: 0.694222
[569]	valid_0's binary_logloss: 0.694579
[570]	valid_0's binary_logloss: 0.694563
[571]	valid_0's binary_logloss: 0.694549
[572]	valid_0's binary_logloss: 0.694634
[573]	valid_0's binary_logloss: 0.694622
[574]	valid_0's binary_logloss: 0.694845
[575]	valid_0's binary_logloss: 0.695163
[576]	valid_0's binary_logloss: 0.695217
[577]	valid_0's binary_logloss: 0.695441
[578]	valid_0's binary_logloss: 0.695654
[579]	valid_0's binary_logloss: 0.695879
[580]	valid_0's binary_logloss: 0.695856
[581]	valid_0's binary_logloss: 0.696118
[582]	valid_0's binary_logloss: 0.696344
[583]	valid_0's binary_logloss: 0.696664
[584]	valid_0's binary_logloss: 0.6967
[585]	valid_0's binary_logloss: 0.697096
[586]	valid_0's binary_logloss: 0.697332
[587]	valid_0's binary_logloss: 0.697424
[588]	valid_0's binary_logloss: 0.697566
[589]	valid_0's binary_logloss: 0.6979
[590]	valid_0's binary_logloss: 0.698053
[591]	valid_0's bina

[350]	valid_0's binary_logloss: 0.635019
[351]	valid_0's binary_logloss: 0.635348
[352]	valid_0's binary_logloss: 0.635568
[353]	valid_0's binary_logloss: 0.635714
[354]	valid_0's binary_logloss: 0.63587
[355]	valid_0's binary_logloss: 0.636201
[356]	valid_0's binary_logloss: 0.636285
[357]	valid_0's binary_logloss: 0.636202
[358]	valid_0's binary_logloss: 0.636103
[359]	valid_0's binary_logloss: 0.636219
[360]	valid_0's binary_logloss: 0.636695
[361]	valid_0's binary_logloss: 0.636714
[362]	valid_0's binary_logloss: 0.636923
[363]	valid_0's binary_logloss: 0.637304
[364]	valid_0's binary_logloss: 0.637511
[365]	valid_0's binary_logloss: 0.637425
[366]	valid_0's binary_logloss: 0.637374
[367]	valid_0's binary_logloss: 0.637591
[368]	valid_0's binary_logloss: 0.637723
[369]	valid_0's binary_logloss: 0.637318
[370]	valid_0's binary_logloss: 0.637308
[371]	valid_0's binary_logloss: 0.637436
[372]	valid_0's binary_logloss: 0.637382
[373]	valid_0's binary_logloss: 0.637157
[374]	valid_0's b

[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.691992
[2]	valid_0's binary_logloss: 0.690468
[3]	valid_0's binary_logloss: 0.688954
[4]	valid_0's binary_logloss: 0.687617
[5]	valid_0's binary_logloss: 0.686182
[6]	valid_0's binary_logloss: 0.684828
[7]	valid_0's binary_logloss: 0.683466
[8]	valid_0's binary_logloss: 0.682175
[9]	valid_0's binary_logloss: 0.68089
[10]	valid_0's binary_logloss: 0.679771
[11]	valid_0's binary_logloss: 0.678485
[12]	valid_0's binary_logloss: 0.677288
[13]	valid_0's binary_logloss: 0.676118
[14]	valid_0's binary_logloss: 0.675125
[15]	valid_0's binary_logloss: 0.673916
[16]	valid_0's binary_logloss: 0.672919
[17]	valid_0's binary_logloss: 0.671739
[18]	valid_0's binary_logloss: 0.670596
[19]	valid_0's binary_logloss: 0.669597
[20]	valid_0's binary_logloss: 0.668453
[21]	valid_0's binary_logloss: 0.667636
[22]	valid_0's binary_logloss: 0.66671
[23]	valid_0's binary_logloss: 0.665646
[24]	valid_0's binary_logloss: 0.66476


[239]	valid_0's binary_logloss: 0.641349
[240]	valid_0's binary_logloss: 0.641109
[241]	valid_0's binary_logloss: 0.641172
[242]	valid_0's binary_logloss: 0.641057
[243]	valid_0's binary_logloss: 0.641235
[244]	valid_0's binary_logloss: 0.641316
[245]	valid_0's binary_logloss: 0.641297
[246]	valid_0's binary_logloss: 0.641371
[247]	valid_0's binary_logloss: 0.641262
[248]	valid_0's binary_logloss: 0.641085
[249]	valid_0's binary_logloss: 0.641142
[250]	valid_0's binary_logloss: 0.641046
[251]	valid_0's binary_logloss: 0.641139
[252]	valid_0's binary_logloss: 0.64095
[253]	valid_0's binary_logloss: 0.641004
[254]	valid_0's binary_logloss: 0.640981
[255]	valid_0's binary_logloss: 0.640944
[256]	valid_0's binary_logloss: 0.640946
[257]	valid_0's binary_logloss: 0.64095
[258]	valid_0's binary_logloss: 0.640709
[259]	valid_0's binary_logloss: 0.640607
[260]	valid_0's binary_logloss: 0.640753
[261]	valid_0's binary_logloss: 0.64064
[262]	valid_0's binary_logloss: 0.640706
[263]	valid_0's bin

[537]	valid_0's binary_logloss: 0.653578
[538]	valid_0's binary_logloss: 0.653867
[539]	valid_0's binary_logloss: 0.653876
[540]	valid_0's binary_logloss: 0.653831
[541]	valid_0's binary_logloss: 0.653696
[542]	valid_0's binary_logloss: 0.653754
[543]	valid_0's binary_logloss: 0.653865
[544]	valid_0's binary_logloss: 0.654117
[545]	valid_0's binary_logloss: 0.65402
[546]	valid_0's binary_logloss: 0.654218
[547]	valid_0's binary_logloss: 0.654323
[548]	valid_0's binary_logloss: 0.654398
[549]	valid_0's binary_logloss: 0.65443
[550]	valid_0's binary_logloss: 0.65453
[551]	valid_0's binary_logloss: 0.654589
[552]	valid_0's binary_logloss: 0.654496
[553]	valid_0's binary_logloss: 0.654574
[554]	valid_0's binary_logloss: 0.654552
[555]	valid_0's binary_logloss: 0.65472
[556]	valid_0's binary_logloss: 0.654733
[557]	valid_0's binary_logloss: 0.654864
[558]	valid_0's binary_logloss: 0.654952
[559]	valid_0's binary_logloss: 0.655146
[560]	valid_0's binary_logloss: 0.655281
[561]	valid_0's bina

[809]	valid_0's binary_logloss: 0.669596
[810]	valid_0's binary_logloss: 0.669464
[811]	valid_0's binary_logloss: 0.669266
[812]	valid_0's binary_logloss: 0.669183
[813]	valid_0's binary_logloss: 0.669285
[814]	valid_0's binary_logloss: 0.669309
[815]	valid_0's binary_logloss: 0.669468
[816]	valid_0's binary_logloss: 0.669386
[817]	valid_0's binary_logloss: 0.669405
[818]	valid_0's binary_logloss: 0.669271
[819]	valid_0's binary_logloss: 0.66911
[820]	valid_0's binary_logloss: 0.669339
[821]	valid_0's binary_logloss: 0.669311
[822]	valid_0's binary_logloss: 0.669279
[823]	valid_0's binary_logloss: 0.66951
[824]	valid_0's binary_logloss: 0.669345
[825]	valid_0's binary_logloss: 0.669647
[826]	valid_0's binary_logloss: 0.669517
[827]	valid_0's binary_logloss: 0.66933
[828]	valid_0's binary_logloss: 0.669385
[829]	valid_0's binary_logloss: 0.669355
[830]	valid_0's binary_logloss: 0.66915
[831]	valid_0's binary_logloss: 0.669173
[832]	valid_0's binary_logloss: 0.669095
[833]	valid_0's bina

[317]	valid_0's binary_logloss: 0.922605
[318]	valid_0's binary_logloss: 0.924129
[319]	valid_0's binary_logloss: 0.924876
[320]	valid_0's binary_logloss: 0.926089
[321]	valid_0's binary_logloss: 0.926551
[322]	valid_0's binary_logloss: 0.927021
[323]	valid_0's binary_logloss: 0.928476
[324]	valid_0's binary_logloss: 0.929106
[325]	valid_0's binary_logloss: 0.931142
[326]	valid_0's binary_logloss: 0.932511
[327]	valid_0's binary_logloss: 0.933554
[328]	valid_0's binary_logloss: 0.936645
[329]	valid_0's binary_logloss: 0.936287
[330]	valid_0's binary_logloss: 0.937664
[331]	valid_0's binary_logloss: 0.939766
[332]	valid_0's binary_logloss: 0.939221
[333]	valid_0's binary_logloss: 0.939477
[334]	valid_0's binary_logloss: 0.939288
[335]	valid_0's binary_logloss: 0.938977
[336]	valid_0's binary_logloss: 0.940026
[337]	valid_0's binary_logloss: 0.940229
[338]	valid_0's binary_logloss: 0.942329
[339]	valid_0's binary_logloss: 0.942724
[340]	valid_0's binary_logloss: 0.944815
[341]	valid_0's 

[197]	valid_0's binary_logloss: 0.677613
[198]	valid_0's binary_logloss: 0.677437
[199]	valid_0's binary_logloss: 0.677827
[200]	valid_0's binary_logloss: 0.677765
[201]	valid_0's binary_logloss: 0.678065
[202]	valid_0's binary_logloss: 0.678115
[203]	valid_0's binary_logloss: 0.678035
[204]	valid_0's binary_logloss: 0.67799
[205]	valid_0's binary_logloss: 0.679022
[206]	valid_0's binary_logloss: 0.678545
[207]	valid_0's binary_logloss: 0.67903
[208]	valid_0's binary_logloss: 0.679476
[209]	valid_0's binary_logloss: 0.679432
[210]	valid_0's binary_logloss: 0.67898
[211]	valid_0's binary_logloss: 0.679662
[212]	valid_0's binary_logloss: 0.679636
[213]	valid_0's binary_logloss: 0.679238
[214]	valid_0's binary_logloss: 0.679343
[215]	valid_0's binary_logloss: 0.679589
[216]	valid_0's binary_logloss: 0.679803
[217]	valid_0's binary_logloss: 0.679768
[218]	valid_0's binary_logloss: 0.680543
[219]	valid_0's binary_logloss: 0.680669
[220]	valid_0's binary_logloss: 0.681397
[221]	valid_0's bin

In [29]:
best_hyperparameters = bayesian_opt.max
best_hyperparameters['params'] = {param: int(value) if param in [
    'depth', 'iterations', 'n_estimators'] else value for param, value in best_hyperparameters['params'].items()}
print("Best hyperparameters:", best_hyperparameters['params'])
print(f"Best R-squared Score: {best_hyperparameters['target']:.4f}")

Best hyperparameters: {'learning_rate': 0.0796548685784298, 'max_depth': 8.16293101479415, 'n_estimators': 159}
Best R-squared Score: 0.6765


In [30]:
params = {
    'max_depth': 8,            
    'learning_rate': 0.0796548685784298,  
    'n_estimators': 159
}

model_LGBM = LGBMClassifier(**params, random_state=27)
model_LGBM.fit(x_train_LGBM, y_train_LGBM)
# предсказание ответов для тренеровочной выборки
y_pred_LGBM_train = model_LGBM.predict(x_train_LGBM)
# предсказание ответов для тестовой выборки
y_pred_LGBM = model_LGBM.predict(x_valid_LGBM)

print("Train LGBM f1_score: ", f1_score(y_train_LGBM, y_pred_LGBM_train))
print("Train LGBM accuracy_score: ", accuracy_score(y_train_LGBM, y_pred_LGBM_train))
print("Validation LGBM f1_score: ", f1_score(y_valid_LGBM, y_pred_LGBM))
print("Validation LGBM accuracy_score: ", accuracy_score(y_valid_LGBM, y_pred_LGBM))

Train LGBM f1_score:  0.9472329472329473
Train LGBM accuracy_score:  0.9471649484536082
Validation LGBM f1_score:  0.6764705882352942
Validation LGBM accuracy_score:  0.6615384615384615


После подбора гиперпараметров мы улучшили метрики и снизили переобучение для модели LGBMClassifier на вадидации f1_score: 0.676, accuracy_score: 0.661.

**Выводы**

Мы сравнили 4 регрессионые модели: DecisionTreeRegressor, CatBoostRegressor, LGBMRegressor, XGBRegressor. Наилучшие результаты показали модели LGBMClassifier с результатом на трейне: f1_score= 0.95 и accuracy_score= 0.95 и тесте: f1_score=0.676, accuracy_score=0.661 и XGBClassifier с результатом на трейне: f1_score= 0.95 и accuracy_score= 0.95 и тесте: f1_score=0.674, accuracy_score=0.677.

**Рекомендации**

Для улучшения значения метрики и снижения переобучения нужно будет увеличить количество данных.